cmd/embedded-cluster/metadata.go

```diff
@@ -65,7 +83,8 @@ func gatherVersionMetadata(k0sCfg *k0sconfig.ClusterConfig) (*types.ReleaseMetad
 		versionsMap[defaults.BinaryName()] = channelRelease.VersionLabel
 	}
 
-	sha, err := goods.K0sBinarySHA256()
+	materializer := goods.NewMaterializer(provider)
+	sha, err := materializer.K0sBinarySHA256()
```

Comment:
this should not require using the provider
there shouldn't be any need to write out the k0s binary, no?

Reply:
yeah, the current code doesn't require writing out the binary at all - so we should just move the function not to depend on a materializer object

Reply:
ok, fine...

Reply:
to clarify, the implementation of `materializer.K0sBinarySHA256()` does not use the `materializer` member functions/variables/etc at all, and so it could be a standalone function and thus not require us to initialize a provider etc

Reply:
https://github.com/replicatedhq/embedded-cluster/blob/80952f3da30129f4fa5c29711303a3e0842918b4/pkg/goods/materializer.go#L193

Reply:
done